msbuild/Xamarin.iOS.Tasks.Core/Tasks/DetectDebugNetworkConfigurationTaskBase.cs

```diff
@@ -48,12 +49,25 @@ public override bool Execute ()
 					hosts = DebuggerHosts.Split (new [] { ';' }, StringSplitOptions.RemoveEmptyEntries);
 
 				if (hosts == null || hosts.Length == 0) {
-					try {
-						ips.AddRange (Dns.GetHostEntry (Dns.GetHostName ()).AddressList.Select ((v) => v.ToString ()));
-					} catch {
-						Log.MTError (7001, "Could not resolve host IPs for WiFi debugger settings.");
-						return false;
-					}
+					var properties = IPGlobalProperties.GetIPGlobalProperties ();
+					var hostName = properties.HostName;
+					IPHostEntry entry;
+
+					do {
+						try {
+							entry = Dns.GetHostEntry (hostName);
+							break;
+						} catch {
+							if (hostName == "localhost") {
+								Log.MTError (7001, "Could not resolve host IPs for WiFi debugger settings.");
+								return false;
+							}
+
+							hostName = "localhost";
```

Comment:
`localhost` will resolve to `127.0.0.1`, which will make the device try to connect to any IDEs running on the device itself, which will obviously not work. The code needs to find the external IP of the device, not localhost.

Reply:
Hmmm, good catch. Any suggestions?

Reply:
I don't know 😒

Exactly what is the network configuration you're having problems with?